x-pack/plugin/profiling/src/test/java/org/elasticsearch/xpack/profiling/HostMetadataTests.java

```diff
@@ -38,4 +37,52 @@ public void testCreateFromRegularSource() {
         assertEquals(region, host.instanceType.region);
         assertEquals(instanceType, host.instanceType.name);
     }
+
+    public void testCreateFromSourceGCP() {
+        final String hostID = "1440256254710195396";
+        final String machine = "x86_64";
+        final String provider = "gcp";
+        final String region = "europe-west1";
+        final String zone = "projects/123456789/zones/" + region + "-b";
```

Comment:
Can we add three more tests here?

* Field contains random garbage
* Region name without hyphen. E.g. `europe1`
* Region name with more hyphens than expected: E.g. `europe-north-west-zone-4`